src/core/data.test.js

```diff
@@ -0,0 +1,222 @@
+const Data = require("./data");
+
+describe("# Data", () => {
+  test("instanciate the data with default options", () => {
+    const options = {};
+    const data = new Data(options);
+    const expectedOptions = {
+      startSymbol: "{{",
+      endSymbol: "}}"
+    };
+    expect(data.options).toEqual(expectedOptions);
+  });
+
+  test("instanciate the data with default options (null)", () => {
+    const data = new Data();
+    const expectedOptions = {
+      startSymbol: "{{",
+      endSymbol: "}}"
+    };
+    expect(data.options).toEqual(expectedOptions);
+  });
+
+  test("instanciate the data using a different symbols", () => {
+    const options = {
+      startSymbol: "{[",
+      endSymbol: "]}"
+    };
+    const data = new Data(options);
+    const expectedOptions = {
+      startSymbol: "{[",
+      endSymbol: "]}"
+    };
+    expect(data.options).toEqual(expectedOptions);
+  });
+
+  test("Throw an error when trying to instanciate if the startSymbole is that more than characters", () => {
+    expect(() => {
+      const options = {
+        startSymbol: "{[[",
+        endSymbol: "]}"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The startSymbol should contains 2 charaters");
+
+    expect(() => {
+      const options = {
+        startSymbol: "[",
+        endSymbol: "]}"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The startSymbol should contains 2 charaters");
+  });
+
+  test("Throw an error when trying to instanciate if the endSymbole is that more than characters", () => {
+    expect(() => {
+      const options = {
+        startSymbol: "{[",
+        endSymbol: "]}}"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The endSymbol should contains 2 charaters");
+
+    expect(() => {
+      const options = {
+        startSymbol: "{[",
+        endSymbol: "]"
+      };
+      // eslint-disable-next-line
+      const data = new Data(options);
+    }).toThrow("The endSymbol should contains 2 charaters");
+  });
+
+  test("add/get values to the datastore (string)", () => {
+    const options = {};
+    const data = new Data(options);
+    data.set("foo", "bar");
+    expect(data.get("foo")).toEqual("foo");
+    expect(data.get("{{foo")).toEqual("{{foo");
+    expect(data.get("{{ foo}}")).toEqual("bar");
+    expect(data.get("{{ foo }}")).toEqual("bar");
+    expect(data.get("{{foo }}")).toEqual("bar");
+    expect(data.get("{{foo}}")).toEqual("bar");
+    expect(data.get("{{foo o }}")).toEqual("{{foo o }}");
+  });
```

Comment:
We probably have two cases there. 
- when the key exists we get the value
- when the key didn't exist (or match) we got the key back.

Or at least we can use named expected results to clarify:
```suggestion
  test("add/get values to the datastore (string)", () => {
    const options = {};
    const data = new Data(options);
    const key = "foo";
    const value = "bar";
  
    data.set(key, value);
  
    // Return back the key passed in args
    const wrongKey = "foo";
    expect(data.get(wrongKey)).toEqual(wrongKey);
    const wrongKey2 = "{{foo";
    expect(data.get(wrongKey2)).toEqual(wrongKey2);
    // on so on ...
    expect(data.get("{{foo o }}")).toEqual("{{foo o }}");
    
    // Good cases
    expect(data.get("{{ foo}}")).toEqual(value);
    expect(data.get("{{ foo }}")).toEqual(value);
    expect(data.get("{{foo }}")).toEqual(value);
    expect(data.get("{{foo}}")).toEqual(value);
  });
```